File: CodingDojo/bomberman-servers/cpp/bomberman/utils/LengthToXY.cpp
#include "LengthToXY.h"


LengthToXY::LengthToXY(int _boardSize) : boardSize(_boardSize) {
}

Point LengthToXY::getXY(int length) const {
	if (length < 0) {
		throw std::invalid_argument("LengthToXY::getXY(int length): length cant be <= 0!");
	}
	return Point(length % boardSize, length / boardSize);
}

int LengthToXY::getLength(int _x, int _y) const {
	return _y * boardSize + _x;
}

File: CodingDojo/games/loderunner/src/main/java/com/codenjoy/dojo/loderunner/client/utils/LengthToXY.java
package com.codenjoy.dojo.loderunner.client.utils;

/**
 * User: oleksandr.baglai
 */
public class LengthToXY {
    private int boardSize;

    public LengthToXY(int boardSize) {
        this.boardSize = boardSize;
    }

    public Point getXY(int length) {
        if (length == -1) {
            return null;
        }
        return new Point(length % boardSize, length / boardSize);
    }

    public int getLength(int x, int y) {
        return (y)*boardSize + x;
    }
}